src/app/beer_garden/api/http/handlers/v1/event.py

```diff
@@ -1,18 +1,62 @@
 # -*- coding: utf-8 -*-
 import logging
+from json import JSONDecodeError
+from typing import TYPE_CHECKING, Optional
 
-from brewtils.errors import RequestForbidden
-from tornado.web import HTTPError
+from brewtils.schema_parser import SchemaParser
+from marshmallow import Schema, ValidationError, fields, validate
 from tornado.websocket import WebSocketHandler
 
+from beer_garden import config
+from beer_garden.api.http.authentication import decode_token, get_user_from_token
+from beer_garden.authorization import user_has_permission_for_object
+from beer_garden.errors import ExpiredTokenException, InvalidTokenException
+
+if TYPE_CHECKING:
+    from brewtils.models import Event
+
+    from beer_garden.db.mongo.models import User
+
+
 logger = logging.getLogger(__name__)
 
 
+def _auth_enabled():
+    """Helper for checking the auth.enabled settings"""
+    return config.get("auth").enabled
+
+
+def _user_can_receive_messages_for_event(user: "User", event: "Event"):
+    """Check a that a user has access to view the supplied event"""
+    user_permitted = True
+    payload_type = event.payload_type.lower()
+
+    if event.payload and not user_has_permission_for_object(
+        user, f"{payload_type}:read", event.payload
+    ):
+        user_permitted = False
+
+    return user_permitted
```

Comment:
Could this whole thing just be
```python
return event.payload and not user_has_permission_for_object(
        user, f"{event.payload_type.lower()}:read", event.payload
    )
```
or is the intention to actually return `True` when the event doesn't have a payload?

Reply:
Returning true when there's no payload is intentional.  Basically if there's no payload then I treat the event as safe for everyone to receive.